common/workunit/workunit.cpp

```diff
@@ -5462,6 +5490,8 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 }
                 if (!iter)
                     return NULL;
+
+                updateWUState(iter);
```

Comment:
You should do this after sortElements which filters based on >filter-lo <filter-hi,
otherwise it's potentially checking a lot of workunits that it doesn't ever return.